tree/ntuple/v7/src/RFieldVisitor.cxx

```diff
@@ -254,11 +254,19 @@ void ROOT::Experimental::RPrintValueVisitor::VisitUInt64Field(const RField<std::
    fOutput << *fValue.Get<std::uint64_t>();
 }
 
-void ROOT::Experimental::RPrintValueVisitor::VisitCardinalityField(const RField<RNTupleCardinality> &field)
+void ROOT::Experimental::RPrintValueVisitor::VisitCardinalityField(const RCardinalityField &field)
 {
    PrintIndent();
    PrintName(field);
-   fOutput << static_cast<std::size_t>(*fValue.Get<RNTupleCardinality>());
+   if (field.Is32Bit()) {
+      fOutput << *fValue.Get<std::uint32_t>();
+      return;
+   }
+   if (field.Is64Bit()) {
+      fOutput << *fValue.Get<std::uint64_t>();
+      return;
+   }
+   fOutput << "\"unsupported cardinality size type\"";
```

Comment:
Could this happen in practice, also given that the `struct RNTupleCardinality` template [cannot be instantiated for any other type](https://github.com/root-project/root/blob/8f3cd2960e231a12311cefea5c0a1fda47171fad/tree/ntuple/v7/inc/ROOT/RNTupleUtil.hxx#L70)?  I am wondering whether we should change that into an assertion, as in
```suggestion
   R__ASSERT(false && "unsupported cardinality size type");
```